### ISSUE TITLE
Provide a default global manifest with no registry and sane defaults for `options.**’ members.

### ISSUE THREAD

**Issue by mkenigs** [flox/flox#408]
- `flox init` should extend the user’s global manifest as its base, adding only `install` and `hook` fields.

Use the following template as the default _global manifest_ saved under `$XDG_CONFIG_HOME/flox/global-manifest.toml`:

There's no need to duplicate structures. Id just say that if the file doesn't exist create it from a hard coded template file.

```toml
[options.allow]
# Whether non-FOSS packages should appear in search/install results.
unfree = true

# Whether "broken" packages should appear in search/install results.
# These are packages with known build/runtime issues.
broken = false

# Limit search/install results to only packages which
# certain SPDX Identifiers by listing them below.
# Leaving `licenses' undefined allows any license to be used.
# See all valid license ids here: https://spdx.org/licenses/
## licenses = ["LGPL2-or-later"]

[options.semver]
# Whether unstable/pre-release versions of software should be
# preferred over the latest stable release in search/install results.
prefer-pre-release = false
```

When initializing a new project environment, extend the users _global manifest_ by appending the following lines ( adding `hook` and `install` ):
```toml
# List packages you wish to install in your environment under
# the `[install]` table
## [install] 
## hello = {}
## nodejs = { version = "^16.4.2" } 

# Set your activation hook ( run when entering the environment )
# You can write this inline with the `script' field, or provide a
# relative path to a file using the `file' field.
[hook]
script = """
  echo "Welcome to your flox environment!";
"""
```

**Comment by mkenigs** [comment]
@aakropotkin I'm trying to split this into steps I understand...is this correct?
- at startup, flox should look for a file in `{flox.config_dir}/global-manifest.toml`. If it doesn't exist, we should create an instance of `GlobalManifest[Raw]` with sane defaults and write it to that file
- `flox init` should extend `global-manifest.toml` with `install` and `hook` fields

Is this all supposed to be on the Rust side though? Are we wanting to duplicate structs like `GlobalManifest[Raw]`? Or would `flox init` start invoking `pkgdb`?

**Comment by aakropotkin** [comment]
@mkenigs that all sounds correct.

This can be handled on either the C++ or Rust side.
Rust seems most sensible here since `pkgdb` doesn't modify files except for its own databases.

**Comment by aakropotkin** [comment]
There's no need to duplicate structures. Id just say that if the file doesn't exist create it from a hard coded template file.

Adding `hook` and `install` is simply appending:
```
# [install] 
# hello =... 
#
# [hook]
# script = """
#   echo Hello;
# """
```

To the global manifest file

**Comment by zmitchell** [comment]
Closed by flox#433
